src/Utilities/VariableTemplates/flattened_tup_chain.jl

```diff
@@ -33,3 +34,54 @@ function flattened_tup_chain(
     Iterators.flatten |>
     collect
 end
+flattened_tup_chain(::AbstractVars{S}) where {S} = flattened_tup_chain(S)
+
+export flattened_named_tuple
+"""
+    flattened_named_tuple(v::AbstractVars)
+
+A flattened NamedTuple, given a `Vars` instance.
+
+# Example:
+
+```julia
+using Test
+using ClimateMachine.VariableTemplates
+nt = (x = 1, a = (y = 2, z = 3, b = ((a = 1,), (a = 2,), (a = 3,))));
+fnt = flattened_named_tuple(nt);
+@test keys(fnt) == (:x, :a_y, :a_z, :a_b_1_a, :a_b_2_a, :a_b_3_a)
```

Comment:
This is demonstrating support for a regular tuple mixed in with `NamedTuple`s? But `Vars` doesn't have those, does it?

Reply:
Ah, EDMF does not use names for the different updrafts?

Reply:
> But Vars doesn't have those, does it?

It does, updrafts are indexed as tuples.